packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,469 @@
+      .toLowerCase();
+
+    let dataSourceClassName = this.artifactInfo.dataSourceClass.replace(
+      'Datasource',
+      'DataSource',
+    );
+
+    Object.assign(this.artifactInfo, {
+      dataSourceClassName: dataSourceClassName,
+    });
+
+    Object.assign(this.artifactInfo, {
+      dataSourceName: dataSourceName,
+    });
+  }
+
+  /**
+   * load the connectors available and check if the basedModel matches any
+   * connectorType supplied for the given connector name
+   * @param {string} connectorType single or a comma separated string array
+   */
+  _isConnectorOfType(connectorType, dataSourceClass) {
+    debug(`calling isConnectorType ${connectorType}`);
+    let jsonFileContent = '';
+    let result = false;
+
+    if (!dataSourceClass) {
+      return false;
+    }
+    let datasourceJSONFile = path.join(
+      this.artifactInfo.datasourcesDir,
+      dataSourceClass.replace('Datasource', '.datasource.json').toLowerCase(),
+    );
+
+    try {
+      jsonFileContent = this.fs.readJSON(datasourceJSONFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${datasourceJSONFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    for (let connector of Object.values(connectors)) {
+      const matchedConnector =
+        jsonFileContent.connector === connector.name ||
+        jsonFileContent.connector === `loopback-connector-${connector.name}`;
+
+      if (matchedConnector && connectorType.includes(connector.baseModel)) {
+        result = true;
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  _setupGenerator() {
+    this.artifactInfo = {
+      type: 'repository ',
+      rootDir: utils.sourceRootDir,
+    };
+
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.repositoriesDir,
+    );
+    this.artifactInfo.datasourcesDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.datasourcesDir,
+    );
+    this.artifactInfo.modelDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.modelsDir,
+    );
+
+    // to be able to write multiple files to the index.ts
+    this.artifactInfo.indexesToBeUpdated = [];
+
+    this.artifactInfo.defaultTemplate = REPOSITORY_CRUD_TEMPLATE;
+
+    this.option('model', {
+      type: String,
+      required: false,
+      description: 'A valid model name',
+    });
+
+    this.option('id', {
+      type: String,
+      required: false,
+      description: 'A valid ID property name for the specified model',
+    });
+
+    this.option('datasource', {
+      type: String,
+      required: false,
+      description: 'A valid datasource name',
+    });
+
+    return super._setupGenerator();
+  }
+
+  setOptions() {
+    return super.setOptions();
+  }
+
+  async checkPaths() {
+    // check for datasources
+    if (!fs.existsSync(this.artifactInfo.datasourcesDir)) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_DIRECTORY} ${this.artifactInfo.datasourcesDir}.
```

Comment:
I think a better message could be `No datasources found at ${this.artifactInfo.datasourcesDir}`.

Reply:
I changed it for both models and repositories. @raymondfeng  PTAL